src/drivers/fpga/xrt/mgmt/xmgmt-main-region.c

```diff
@@ -315,10 +319,25 @@ static int xmgmt_region_program(struct fpga_region *re, const void *xclbin)
 	if (re->get_bridges)
 		fpga_bridges_put(&re->bridge_list);
 
```

Comment:
I think it needs to return if program failed.

Reply:
Good catch! fixed.